src/main/java/io/github/christophermanahan/carnitas/HTTPResponse.java

```diff
@@ -26,20 +30,34 @@
         this.status = status;
     }
 
-    private HTTPResponse(Status status, Optional<String> body) {
+    private HTTPResponse(Status status, List<String> headers, Optional<String> body) {
         this.status = status;
+        this.headers = headers;
         this.body = body;
     }
 
+    HTTPResponse withHeaders(List<String> headers) {
```

Comment:
I think I'm still a bit confused as to why you would use `withHeaders` and `withBody`. They're just creating a `new HTTPResponse` so wouldn't you just call `new HTTPResponse(status, headers, body)` directly?

Reply:
I am doing this because the body and headers are optional and only need to be added in certain scenarios.

Reply:
I'm thinking something like this:
```
 HTTPResponse(Status status) {
        this.status = status;
    }

    HTTPResponse(Status status, List<String> headers) {
        this.status = status;
        this.headers = headers;
    }

    HTTPResponse(Status status, List<String> headers, Optional<String> body) {
        this.status = status;
        this.headers = headers;
        this.body = body;
    }
```

```
       ArrayList headers = new ArrayList<String>();
        new HTTPResponse(
          HTTPResponse.Status.OK,
          headers
        );
```

I felt like `withHeaders` and `withBody` were just adding some unnecessary complexity. As you can see now there's 3 constructors. There might even be a pattern that cleans up this class even further.